packages/pangea-sdk/.sdk-ci.yml

```diff
@@ -80,72 +84,71 @@
         ENV: ${SERVICE_VAULT_ENV}
         TEST: test_vault
   rules:
-    - if: '$CLOUD == "GCP" && $TEST == "test_file_scan"'
+    - if: $CI_COMMIT_BRANCH && '$CLOUD == "GCP" && $TEST == "test_file_scan"'
       allow_failure: true
-    - if: '$CLOUD == "GCP" && $TEST != "test_file_scan"'
+    - if: $CI_COMMIT_BRANCH && '$CLOUD == "GCP" && $TEST != "test_file_scan"'
       allow_failure: false
-    - if: '$CLOUD != "GCP"'
+    - if: $CI_COMMIT_BRANCH && '$CLOUD != "GCP"'
       allow_failure: false
 
-.python-sdk-publish-base:
-  before_script:
-    - cd packages/pangea-sdk/
-    - pip install poetry
-    - poetry check
-    - poetry install
-  rules:
-    - if: $CI_COMMIT_BRANCH == "release"
-      changes:
-        - packages/pangea-sdk/**/*
-      when: on_success
-
-python-sdk-lint:
-  extends: .python-sdk-base
+sdk-lint-black:
+  extends: .sdk-base
   stage: lint
+  needs: []
   script:
     - poetry run black .
 
-python-sdk-generate-docs:
-  extends: .python-sdk-base
+sdk-lint-mypy:
+  extends: .sdk-base
   stage: lint
-  allow_failure: true
+  needs: []
+  script:
+    - poetry run mypy pangea tests
+
+sdk-build:
+  extends: .sdk-base
+  stage: build
+  script:
+    - poetry build
+  artifacts:
+    paths: ["packages/pangea-sdk/dist"]
+    when: on_success
+
+sdk-docs:
+  extends: .sdk-base
+  stage: build
   script:
     - poetry run python parse_module.py > python_sdk.json
   artifacts:
     expire_in: 1 month
     expose_as: python_sdk
-    paths: ['packages/pangea-sdk/python_sdk.json']
+    paths: ["packages/pangea-sdk/python_sdk.json"]
     when: on_success
 
-python-sdk-type-check:
-  extends: .python-sdk-base
-  stage: lint
-  needs: []
-  script:
-    - poetry run mypy pangea tests
-
-python-sdk-unit-tests:
-  extends: .python-sdk-base
+sdk-unit-tests:
+  extends: .sdk-base
   stage: unit_tests
   script:
     - poetry run python -m unittest tests.unit
 
-python-sdk-integration-tests:
-  extends: .python-sdk-test-base
+sdk-integration-tests:
+  extends: .sdk-test-base
   stage: integration_tests
   script:
     - poetry run python -m unittest tests.integration.${TEST}
 
 # TODO: Add asyncio tests when upgrade python 3.8
-# python-sdk-asyncio-integration-tests:
-#   extends: .python-sdk-test-base
+# sdk-asyncio-integration-tests:
+#   extends: .sdk-test-base
 #   stage: integration_tests
 #   script:
 #     - poetry run python -m unittest tests.integration.asyncio.${TEST}
 
-python-sdk-publish:
-  extends: .python-sdk-publish-base
+sdk-publish:
+  extends: .sdk-base
   stage: publish
+  needs: [sdk-build]
   script:
-    - poetry build
     - poetry publish --username __token__ --password $PYPI_SDK_TOKEN
```

Comment:
I'm not sure if this is needed since poetry publish will fail, but do you want to 
- do a pre-check in the merge to main pipeline to make sure the pyproject.toml version isn't already in PyPi to save some cycles?
- validate that the tag version made in the repo matches the pyproject.toml version?

these should never happen, but wondering if you want to add the logic for that or let it be

Reply:
Agree with @doubletooth, due to right now release a new version will be only based on a tag push, without a double check (MR review), I'd add 2 checks:
1.  Checking that tag version matches that one in toml file
2. Checking that tags' pushed outside `main` branch has a required postfix and also match version in toml file. 

Also, should we defined postfix format? It should be added to docs somewhere.

Reply:
Added a bash script to validate the Git tag version matches the `pyproject.toml` version, feel free to tear it apart 😁 .

Reply:
@kenany It's okey for me. But again, we should add the check that it's on `main` branch to avoid releases happen from dev branches.
And we should discuss how to publish pre-release versions

Reply:
Git tags are independent of branches (right?), so we can't do such a branch check.